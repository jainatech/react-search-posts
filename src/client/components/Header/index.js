import React from 'react'
import PropTypes from 'prop-types'
import { Navbar, Form, FormControl } from 'react-bootstrap';

function Header(props) {
    const searchRef = React.createRef();

    const doSearch = () => {
        if (searchRef && searchRef.current && searchRef.current.value) {
            props.filterData(searchRef.current.value);
        }
    }

    return (
        <>
            <Navbar expand="lg" variant="light" bg="light" className="bg-light justify-content-between">
                <Navbar.Brand>{props.text}</Navbar.Brand>
                {props.search ?
                    <Form inline>
                        <FormControl ref={searchRef} type="text" placeholder="Search by title" className=" mr-sm-2" onChange={doSearch} />
                    </Form> : null}

            </Navbar>
        </>
    )
}

Header.propTypes = {
    text: PropTypes.string,
    filterData: PropTypes.func
}

export default Header

