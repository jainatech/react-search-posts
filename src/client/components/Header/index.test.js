import React from 'react';
import { mount } from 'enzyme';
import Header from './index';
import { BrowserRouter as Router } from 'react-router-dom';

const createTestProps = (props) => ({
  ...props
})

describe('Header module case--->', () => {
  it('Header match snapshot', () => {
    let props = createTestProps({
        text: 'Hello',
        search: true
    })
    let wrapper = mount(<Router><Header {...props}></Header></Router>);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});