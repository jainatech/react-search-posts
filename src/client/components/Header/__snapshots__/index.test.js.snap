// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header module case---> Header match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header
      search={true}
      text="Hello"
    >
      <Navbar
        bg="light"
        className="bg-light justify-content-between"
        collapseOnSelect={false}
        expand="lg"
        variant="light"
      >
        <nav
          className="bg-light justify-content-between navbar navbar-expand-lg navbar-light bg-light"
        >
          <NavbarBrand>
            <span
              className="navbar-brand"
            >
              Hello
            </span>
          </NavbarBrand>
          <Form
            inline={true}
          >
            <form
              className="form-inline"
            >
              <FormControl
                className=" mr-sm-2"
                onChange={[Function]}
                placeholder="Search by title "
                type="text"
              >
                <input
                  className=" mr-sm-2 form-control"
                  onChange={[Function]}
                  placeholder="Search by title "
                  type="text"
                />
              </FormControl>
            </form>
          </Form>
        </nav>
      </Navbar>
    </Header>
  </Router>
</BrowserRouter>
`;
